src/datepicker/Datepicker.vue

```diff
@@ -1,17 +1,21 @@
 <template>
   <div class="v3dp__datepicker">
-    <input
-      type="text"
-      readonly="readonly"
-      v-model="input"
-      v-bind="$attrs"
-      :placeholder="placeholder"
-      :disabled="disabled"
-      :tabindex="disabled ? -1 : 0"
-      @blur="renderView()"
-      @focus="renderView(startingView)"
-      @click="renderView(startingView)"
-    />
+    <div class="v3dp__input_wrapper">
+      <input
+          type="text"
+          readonly="readonly"
+          v-model="input"
+          :placeholder="placeholder"
+          :disabled="disabled"
+          :tabindex="disabled ? -1 : 0"
+          @blur="renderView()"
+          @focus="renderView(startingView)"
+          @click="renderView(startingView)"
+      />
+      <slot>
```

Comment:
This is better, but I think you missed two things:
1. I think there will be more slots required later down the line, so we should make it named (e.g. `<slot name="clear">`)
2. If you used that slot, you couldn't actually clear anything by clicking, since the function would not be called.
I think this is solvable by providing this function to the [slot scope](https://v3.vuejs.org/guide/component-slots.html#scoped-slots), so that the user could bind the action to their own control, essentially. Other solution could be to wrap the slot in a `<button>` and do `@click` on that. That way we have more control over the action, and the end user can worry only about styling.